api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java

```diff
@@ -27,149 +27,205 @@
+	 *             the {@code com.sap.cloud.security.token} package.
 	 * @return user name
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getLogonName() throws XSUserInfoException;
 
 	/**
 	 * Given name of the user.
-	 * 
+	 *
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.GIVEN_NAME)} from
+	 *             the {@code com.sap.cloud.security.token} package.
 	 * @return given name
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getGivenName() throws XSUserInfoException;
 
 	/**
 	 * Familiy name of the user.
-	 * 
+	 *
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.FAMILY_NAME)}
+	 *             from the {@code com.sap.cloud.security.token} package.
 	 * @return family name
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getFamilyName() throws XSUserInfoException;
 
 	/**
 	 * Return the user origin. The origin is an alias that refers to a user store in
 	 * which the user is persisted. For example, users that are authenticated by the
 	 * UAA itself with a username/password combination have their origin set to the
 	 * value uaa.
-	 *
+	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.ORIGIN)} from the
+	 *             {@code com.sap.cloud.security.token} package.
 	 * @return user origin
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getOrigin() throws XSUserInfoException;
 
 	/**
 	 * Return identity zone
-	 *
+	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBACCOUNT_ID)}
+	 *             from the {@code com.sap.cloud.security.token} package.
 	 * @return identity zone
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	String getIdentityZone() throws XSUserInfoException;
 
 	/**
 	 * Return subaccount identifier
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBACCOUNT_ID)}
+	 *             from the {@code com.sap.cloud.security.token} package.
 	 * @return subaccount identifier
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getSubaccountId() throws XSUserInfoException;
 
 	/**
 	 * Return the subdomain of this subaccount
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBDOMAIN)} from
```

Comment:
this is NOT correct / its an external attribute:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java#L178

Reply:
Right. We have one indirection here. We could implement this though by creating something like getStringAttributeFromClaim which is possible. I correct the docs.